specs/experimental/contracts/cannon/mips64.md

```diff
@@ -0,0 +1,194 @@
+**Table of Contents**
+
+- [Overview](#overview)
+- [Definitions](#definitions)
+  - [Thread Stack](#thread-stack)
+  - [Memory Reservation](#memory-reservation)
+  - [Context Switch](#context-switch)
+  - [State Version](#state-version)
+- [Assumptions](#assumptions)
+  - [a01-001: PreimageOracle provides valid data](#a01-001-preimageoracle-provides-valid-data)
+    - [Mitigations](#mitigations)
+  - [a01-002: Proof data is well-formed](#a01-002-proof-data-is-well-formed)
+    - [Mitigations](#mitigations-1)
+- [Invariants](#invariants)
+  - [i01-001: Active thread stack is never empty during execution](#i01-001-active-thread-stack-is-never-empty-during-execution)
+    - [Impact](#impact)
+  - [i01-002: Memory reservations are mutually exclusive](#i01-002-memory-reservations-are-mutually-exclusive)
+    - [Impact](#impact-1)
+  - [i01-003: Thread stack roots correctly represent thread contents](#i01-003-thread-stack-roots-correctly-represent-thread-contents)
+    - [Impact](#impact-2)
+  - [i01-004: State transitions are deterministic](#i01-004-state-transitions-are-deterministic)
+    - [Impact](#impact-3)
+- [Function Specification](#function-specification)
+  - [step](#step)
+  - [oracle](#oracle)
+  - [stateVersion](#stateversion)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The MIPS64 contract is an on-chain implementation of a MIPS64 virtual machine that executes a single instruction step.
+It serves as the state transition verification component in the Cannon fault proof system, enabling on-chain validation
+of off-chain computation by proving correct execution of MIPS64 instructions within dispute games.
+
+## Definitions
+
+### Thread Stack
+
+A hash onion structure representing a stack of thread states. Each thread is committed by computing
+`Keccak256(previous_root ++ thread_hash)`, where the empty stack is represented by
+`Keccak256(bytes32(0) ++ bytes32(0))`. This construction allows succinct commitment to all threads using a single
+bytes32 value. See the [Cannon Fault Proof VM specification](../../fault-proof/cannon-fault-proof-vm.md#thread-stack-hashing)
+for detailed hashing mechanics.
+
+### Memory Reservation
+
+A mechanism enabling atomic read-modify-write operations through Load Linked and Store Conditional instructions.
+A reservation tracks a memory address, the owning thread, and the operation size (32-bit or 64-bit). Any memory write
+to the reserved address clears the reservation, ensuring atomicity across concurrent threads.
+
+### Context Switch
+
+The process of preempting the currently active thread and activating another thread from the thread stacks.
+Context switches occur voluntarily through specific syscalls or forcibly after executing a quantum of steps without
+yielding.
+
+### State Version
+
+An identifier specifying the state transition rules implemented by the contract. The MIPS64 contract supports
```

Comment:
devinai Remove the specific references to specific state versions that are supported